File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.js
function solution(s){
    const stack = [];

    for(let bracket of s.split("")) {
        if(bracket === "(") stack.push(bracket);
        else {
            if(stack.length === 0) {
                return false;
            } else {
                stack.pop();
            }
        }
    }
    
    return !Boolean(stack.length);
}

File: 프로그래머스/2/12909. 올바른 괄호/올바른 괄호.py
def solution(s):
    brace = []
    
    for c in s:
        if c == "(":
            brace.append(c)
        else:
            if not brace:
                return False
            if brace[-1] == ")":
                return False
            brace.pop()
    
    return True if not brace else False